Add plotting from the console example (#344)
r-wasm/webr: src/docs/plotting.qmd

</script>
```

## Example: Plotting from the console

> The Console class includes callbacks that are used for handling image rendering. This example builds off of the interactive webR REPL Console. In addition to the console, there is a `<canvas>` element which will be plots will be drawn to. The callbacks `canvasImage` and `canvasNewPage` are used to draw plots. 

Reviewer: ```suggestion
The `Console` class includes callbacks that are used for handling image rendering. This example builds off the [interactive webR REPL Console](examples.qmd#creating-an-interactive-webr-repl-console). In addition to the console, there is a `<canvas>` element to which plots will be drawn. The callbacks `canvasImage` and `canvasNewPage` are used to draw plots.
```

> When a new plot is created, the `canvasNewPage` callback is used clearing the bitmap from the canvas using `.reset()`. Subsequently, the `canvasImage` callback is used to draw the `ImageBitMap` object onto the canvas. 

Reviewer: ```suggestion
When a new plot is created, the `canvasNewPage` callback is used clearing the bitmap from the canvas using [`reset()`](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/reset). Subsequently, the `canvasImage` callback is used to draw the `ImageBitMap` object onto the canvas.
```

> ```{=html}

Reviewer: ```suggestion
``` html
```

This displays the html formatted chunk on the page, rather than embedding it directly.

<html>
  <head>
    <title>WebR Test Console</title>
    <style>
      body {
          display: flex;
      }

>       #plot-output, #out {
>           flex: 1;
>       }

Reviewer: ```suggestion
```

I think the resulting example page looks better without this, though it is a personal preference.

    </style>
  </head>
  <body>
    <div id="plot-output">
      <canvas width="500" height="500" id="plot-canvas"></canvas>
    </div>
    <div>
      <pre><code id="out">Loading webR, please wait...</code></pre>
      <input spellcheck="false" autocomplete="off" id="input" type="text">
      <button onclick="globalThis.sendInput()" id="run">Run</button>
    </div>
    
    <script type="module">
      /* Create a webR console using the Console helper class */
      import { Console } from 'https://webr.r-wasm.org/latest/webr.mjs';

      var canvas = document.getElementById("plot-canvas")
      var ctx = canvas.getContext('2d');

      const webRConsole = new Console({
        stdout: line => document.getElementById('out').append(line + '\n'),
        stderr: line => document.getElementById('out').append(line + '\n'),
        prompt: p => document.getElementById('out').append(p),
        canvasImage: ci => ctx.drawImage(ci, 0, 0),
        canvasNewPage: () => ctx.reset(),
      });
      webRConsole.run();

>       // modify the output to be half the size of the canvas

Reviewer: ```suggestion
      /* Set the default graphics device to be half the canvas element size */
```
